Application/Setting/Univention/Service.php

```diff
@@ -0,0 +1,507 @@
+                $UploadItem['firstname'] = '';
+                $UploadItem['lastname'] = '';
+                $UploadItem['record_uid'] = $tblAccount->getId();
+                $UploadItem['source_uid'] = $Acronym.'-'.$tblAccount->getId();
+                $UploadItem['roles'] = '';
+                $UploadItem['schools'] = '';
+                $UploadItem['mail'] = '';
+                $UploadItem['groupArray'] = '';
+
+                $UploadItem['password'] = '';
+//                $UploadItem['password'] = $tblAccount->getPassword(); // ??
+                $UploadItem['school_classes'] = '';
+
+                if ($tblPerson = Account::useService()->getPersonAllByAccount($tblAccount)){
+                    $tblPerson = current($tblPerson);
+                    $UploadItem = $this->getPersonDataExcel($UploadItem, $tblPerson, $tblYear, $Acronym, $TeacherClasses, $TeacherSchools);
+                } else {
+                    // Ohne Person kein sinnvoller Account
+                    continue;
+                }
+
+                if($UploadItem){
+                    array_push($UploadToAPI, $UploadItem);
+                }
+            }
+        }
+        if($StudentWithoutAccount){
+            $tblGroup = Group::useService()->getGroupByMetaTable(TblGroup::META_TABLE_STUDENT);
+            $tblPersonList = Group::useService()->getPersonAllByGroup($tblGroup);
+            if($tblPersonList){
+                foreach($tblPersonList as $tblPerson){
+                    if(Account::useService()->getAccountAllByPerson($tblPerson)){
+                        // ignore students with account
+                        continue;
+                    }
+                    $Item['name'] = '';
+                    $Item['firstname'] = '';
+                    $Item['lastname'] = '';
+                    $Item['record_uid'] = '';
+                    $Item['source_uid'] = $Acronym.'-';
+                    $Item['roles'] = '';
+                    $Item['schools'] = '';
+                    $Item['password'] = '';
+                    $Item['school_classes'] = '';
+                    $Item['mail'] = '';
+                    $Item['groupArray'] = '';
+
+                    $Item = $this->getPersonDataExcel($Item, $tblPerson, $tblYear, $Acronym, $TeacherClasses, $TeacherSchools);
+
+                    if($Item){
+                        array_push($UploadToAPI, $Item);
+                    }
+                }
+            }
+        }
+
+        return (!empty($UploadToAPI) ? $UploadToAPI : false);
+    }
+
+    /**
+     * @param array     $Item
+     * @param TblPerson $tblPerson
+     * @param TblYear   $tblYear
+     * @param string    $Acronym
+     * @param array     $TeacherClasses
+     * @param array     $TeacherSchools
+     *
+     * @return bool|array
+     */
+    private function getPersonDataExcel($Item, TblPerson $tblPerson, TblYear $tblYear, $Acronym, $TeacherClasses, $TeacherSchools)
```

Comment:
Codacy found an issue: [The method getPersonDataExcel() has 126 lines of code. Current threshold is set to 100. Avoid really long methods.](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5757982)